### PR TITLE
[MINOR][DOC]Add additional explanation for "Setting the max receiving rate" in streaming-programming-guide.md

### PR DESCRIPTION
**Pull request by hehuiyuan**
In streaming-programming-guide.md, as follows:

Setting the max receiving rate - If the cluster resources is not large enough for the streaming
application to process data as fast as it is being received, the receivers can be rate limited
by setting a maximum rate limit in terms of records / sec.
See the configuration parameters
spark.streaming.receiver.maxRate for receivers and spark.streaming.kafka.maxRatePerPartition
for Direct Kafka approach. In Spark 1.5, we have introduced a feature called backpressure **_**that
eliminate the need to set this rate limit**_**, as Spark Streaming automatically figures out the
rate limits and dynamically adjusts them if the processing conditions change. This backpressure
can be enabled by setting the configuration parameter
spark.streaming.backpressure.enabled to true.

I think we should be more rigorous. Setting the rate limit when restarting task is very necessary.The first batch may be processing all the time and can not run normally when the first batch of data is very large for Direct Kafka approach .

If we know how the task works ,for example, processing time/maximum amount of data that can be processed per batch, setting a reasonable rate can make the task more stable.

### CODE REVIEW

File: docs/streaming-programming-guide.md [hunk 1]
```diff
@@ -2036,7 +2036,7 @@ To run a Spark Streaming applications, you need to have the following.
   `spark.streaming.receiver.maxRate` for receivers and `spark.streaming.kafka.maxRatePerPartition`
   for Direct Kafka approach. In Spark 1.5, we have introduced a feature called *backpressure* that
   eliminate the need to set this rate limit, as Spark Streaming automatically figures out the
-  rate limits and dynamically adjusts them if the processing conditions change. This backpressure
+  rate limits and dynamically adjusts them if the processing conditions change.If the first batch of data is very large which causes the first batch is processing all the time and the task can not work normally , using a maximum rate limit can solve the problem .This backpressure
```
Comment (srowen) [inline docs/streaming-programming-guide.md]: I don't quite understand this sentence, and it contradicts the previous one, which says it's not needed. I don't think this helps.

Comment (hehuiyuan) [reply docs/streaming-programming-guide.md]: First of all,think you for your reply.

The original document means that setting backpressure does not require to set this rate limit。However, In actual usage scenarios, such as spark streaming consuming kafka, the first batch of data is often very large, leading to the first batch has been processing, affecting the normal operation of tasks。Even the first batch of data is finished and it  costs much more time than the batch time , the efficiency of processing  subsequent batch is not as good as the efficiency of the first batch of data  was processed in batch time then continue  processing ubsequent batch.

In a word,i want to express  setting backpressure is not need setting rate limit that is not rigorous .

Comment (gaborgsomogyi) [reply docs/streaming-programming-guide.md]: I see the intention but I agree with Sean and think this change doesn't make the doc better. I agree that if the first batch processing time is significantly bigger than the batch period then microbatches can be queued up but I would rephrase things.

Comment (hehuiyuan) [reply docs/streaming-programming-guide.md]: Maybe my expression is inappropriate.My main purpose is to provide personal understanding and propose.I believe you can have a better explanation.
